ffi/src/scan.rs

```diff
@@ -32,6 +31,23 @@ pub struct SharedScan;
 #[handle_descriptor(target=ScanMetadata, mutable=false, sized=true)]
 pub struct SharedScanMetadata;
 
+/// A predicate that can be used to skip data when scanning.
+///
+/// When invoking [`scan`], The engine provides a pointer to the (engine's native) predicate,
+/// along with a visitor function that can be invoked to recursively visit the predicate. This
+/// engine state must be valid until the call to [`scan::scan`] returns. Inside that method, the
```

Comment:
now that it's in the same mod i think it can just be:
```suggestion
/// engine state must be valid until the call to [`scan`] returns. Inside that method, the
```